pilot/pkg/autoregistration/controller_test.go

```diff
@@ -48,6 +49,43 @@ func init() {
 	features.WorkloadEntryCleanupGracePeriod = 50 * time.Millisecond
 }
 
+var _ Connection = &fakeConn{}
+
+type fakeConn struct {
```

Comment:
slightly prefer to just make Connection a struct and for the non-fake one just copy the id, proxy, etc into it

Reply:
Do you mean make a struct `autoregistration.Connection` and set it up here: 

 ```
...
	if err := s.WorkloadEntryController.OnConnect(autoregistrion.Connection{
  ID: con.ID,
  Proxy: con.proxy,
  Stop: con.stop,
}); err != nil {
...
```

The reason for using an interface was to avoid cyclic reference.

Reply:
Yes. bummer it causes a cycle